File: snipp.py
def is_palindrome_number(num):
    original = str(num)
    reversed_num = original[::-1]
    return original == reversed_num

# Example usage
number = int(input("Enter a number: "))

if is_palindrome_number(number):
    print(f"{number} is a palindrome.")
else:
    print(f"{number} is not a palindrome.")

File: snipp.java
import java.util.Scanner;

public class PalindromeChecker {

    public static boolean isPalindrome(int number) {
        int original = number;
        int reversed = 0;

        while (number > 0) {
            int digit = number % 10;            // get last digit
            reversed = reversed * 10 + digit;   // build reversed number
            number /= 10;                       // remove last digit
        }

        return original == reversed;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Enter a number: ");
        int input = scanner.nextInt();

        if (isPalindrome(input)) {
            System.out.println(input + " is a palindrome.");
        } else {
            System.out.println(input + " is not a palindrome.");
        }

        scanner.close();
    }
}
